### PR TITLE
SOLR-16392: Minor CollectionsHandler cleanup

### PR DESCRIPTION
**Pull request by gerlowskija**
https://issues.apache.org/jira/browse/SOLR-16392

# Description

CollectionsHandler.handleRequestBody had an unnecessary switch-case statement that cluttered thing up, instead of leaning on the CollectionsOperation enum already present for this purpose.

# Solution

This PR removes the switch-case in favor of using CollectionOperation as was done previously.  Credit for Bruno R for the suggestion.

# Tests

No new tests added for this minor refactor.  Existing tests continue to pass.
# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.

### CODE REVIEW

**Review by bruno-roustant — APPROVED** [review]
LGTM
I understand your point that this CollectionsHandler class is too big, mostly because of the CollectionOperation enum being too big. We could probably move the enum out of CollectionsHandler, but that's for another future PR.

**Comment by gerlowskija** [comment]
Still waiting on some tests running locally, but will merge when those finish up.

**Pull request merged**
